bftengine/src/bftengine/SigManager.cpp

```diff
@@ -230,11 +230,16 @@ void SigManager::sign(const char* data, size_t dataLength, char* outSig, uint16_
 uint16_t SigManager::getMySigLength() const { return (uint16_t)mySigner_->signatureLength(); }
 
 void SigManager::setClientPublicKey(const std::string& key, PrincipalId id, KeyFormat format) {
+  LOG_INFO(KEY_EX_LOG, "client: " << id << " key: " << key << " format: " << (uint16_t)format);
   if (replicasInfo_.isIdOfExternalClient(id)) {
-    std::unique_lock lock(mutex_);
-    verifiers_.insert_or_assign(id, std::make_shared<RSAVerifier>(key.c_str(), format));
+    try {
+      std::unique_lock lock(mutex_);
```

Comment:
in which thread ```setClientPublicKey``` is being called and with which threads does it have a race i.e. why do we need to lock?
and if we have a race doesn't it mean we need to lock in other places that use the ``` verifiers_``` e.g 
 ```SigManager::verifySig```

Reply:
This was @glevkovich input in the previous PR.
In case of a single thread when no contention occurs it's harmless.